src/app/user/components/create-profile/create-profile.component.ts

```diff
@@ -0,0 +1,122 @@
+import { User } from 'firebase';
+import { UserProfile } from 'src/app/shared/models/userProfile';
+
+@Component({
+  selector: 'app-create-profile',
+  templateUrl: './create-profile.component.html',
+  styleUrls: ['./create-profile.component.less']
+})
+export class CreateProfileComponent implements OnInit {
+  companyUserType: UserType;
+  normalUserType: UserType;
+  userType?: UserType;
+  firstName: string;
+  lastName: string;
+  companyName?: string;
+  birthDate: Date;
+  interests?: string;
+  workExperience?: string;
+  educationalExperience?: string;
+  profileFormNormal: FormGroup;
+  profileFormCompany: FormGroup;
+  isFormNomalUserSubmitted: boolean;
+  isFormCompanySubmitted: boolean;
+  currentUser$: AngularFirestoreDocument<User>;
+  id$:string;
+
+  constructor(
+    private formBuilder: FormBuilder,
+    private router: Router,
+    private route: ActivatedRoute,
+    private afs: AngularFirestore
+  ) {
+
+    this.isFormCompanySubmitted = false;
+    this.isFormNomalUserSubmitted = false;
+    this.companyUserType = UserType.CompanyUser;
+    this.normalUserType = UserType.NormalUser;
+    this.profileFormNormal = this.formBuilder.group({
+      firstName: ['', [Validators.required]],
+      lastName: ['', [Validators.required]],
+      birthDate: ['', Validators.required],
+      interests: ['', Validators.maxLength(250)],
+      workExperience: ['', Validators.maxLength(250)],
+      educationalExperience: ['', Validators.maxLength(250)]
+    });
+
+    this.profileFormCompany = this.formBuilder.group({
+      firstName: ['', [Validators.required]],
+      lastName: ['', [Validators.required]],
+      companyName: ['', [Validators.required]],
+      birthDate: ['', [Validators.required]]
+    });
+
+    let sub = this.route.params.subscribe(params => {
+      this.id$ = params['id'];
+    });
+    this.currentUser$ = this.afs.collection<User>('users').doc(this.id$);
+
+    this.currentUser$.ref.get().then(function (doc) {
+      if (!doc.exists) {
+        alert("User not found.");
+        this.router.navigate([`/home`]);
+      }
+    });
+  }
+  selectHire() {
+    this.userType = UserType.CompanyUser;
+  }
+
+  selectFindJob() {
+    this.userType = UserType.NormalUser;
+  }
+  resetUser() {
+    this.userType = undefined;
+  }
+
+  onSubmit() {
+    let userProfile: UserProfile;
+    if (this.userType == UserType.CompanyUser) {
+      this.isFormCompanySubmitted = true;
+      if(this.profileFormCompany.status === 'VALID'){
+        userProfile = {
+          userType: this.userType,
+          firstName: this.profileFormCompany.controls.firstName.value,
+          lastName: this.profileFormCompany.controls.lastName.value,
+          birthDate: this.profileFormCompany.controls.birthDate.value,
+          companyName: this.profileFormCompany.controls.companyName.value
+        }
+          this.currentUser$.ref.update({userProfile:userProfile});
```

Comment:
move this line to a `UserProfileService` in a method like `updateUserProfile(userProfile: UserProfile)`